contracts/liquidityMining/KyberFairLaunch.sol

```diff
@@ -0,0 +1,469 @@
+  * @param _rewardLockBps: percentage (in bps) of reward to be locked
+  * @param _rewardPerBlock: amount of reward token per block for the pool
+  * @param _withUpdate: whether to update all pools
+  */
+  function addPool(
+    address _stakeToken,
+    uint32 _startBlock,
+    uint32 _endBlock,
+    uint32 _rewardLockBps,
+    uint128 _rewardPerBlock,
+    bool _withUpdate
+  ) external override onlyAdmin {
+    if (_withUpdate) {
+      massUpdatePools();
+    }
+    require(_stakeToken != address(0), 'add: not stakeToken addr');
+    require(!isDuplicatedPool(_stakeToken), 'add: duplicated pool');
+
+    require(
+      _startBlock > block.number && _endBlock > _startBlock, 'add: invalid blocks'
+    );
+    require(_rewardLockBps <= BPS, 'add: invalid lock bps');
+    require(_rewardPerBlock > 0, 'add: invalid reward per block');
+
+    poolInfo.push(
+      PoolInfo({
+        stakeToken: _stakeToken,
+        startBlock: _startBlock,
+        endBlock: _endBlock,
+        lastRewardBlock: _startBlock,
+        rewardLockBps: _rewardLockBps,
+        rewardPerBlock: _rewardPerBlock,
+        accRewardPerShare: 0
+      })
+    );
+
+    emit AddNewPool(
+      _stakeToken,
+      _startBlock,
+      _endBlock,
+      _rewardLockBps,
+      _rewardPerBlock
+    );
+  }
+
+  /**
+  * @dev Renew a pool with a new set of data, only when pool is not active.
+  * @param _pid: pool id to be renew
+  * @param _startBlock: block where the reward starts
+  * @param _endBlock: block where the reward ends
+  * @param _rewardLockBps: percentage (in bps) of reward to be locked
+  * @param _rewardPerBlock: amount of reward token per block for the pool
+  * @param _withUpdate: whether to update the pool
+  */
+  function renewPool(
+    uint256 _pid,
+    uint32 _startBlock,
+    uint32 _endBlock,
+    uint32 _rewardLockBps,
+    uint128 _rewardPerBlock,
+    bool _withUpdate
+  ) external override onlyAdmin {
+    require(_pid < poolInfo.length, 'renew: invalid pool id');
+    // only update this pool
+    if (_withUpdate) updatePoolRewards(_pid);
+
+    PoolInfo storage pool = poolInfo[_pid];
+    {
+      // prevent stack too deep
+      uint256 currentBlock = block.number;
+      require(
+        pool.startBlock > currentBlock || pool.endBlock < currentBlock,
+        'renew: pool is active'
+      );
+
+      require(
+        _startBlock > currentBlock && _endBlock > _startBlock, 'renew: invalid blocks'
+      );
+      require(_rewardLockBps <= BPS, 'renew: invalid lock bps');
+      require(_rewardPerBlock > 0, 'renew: invalid reward per block');
+    }
+
+    {
+      // prevent stack too deep
+      (
+        pool.startBlock,
+        pool.endBlock,
+        pool.rewardLockBps,
+        pool.lastRewardBlock
+      ) = (
+        _startBlock,
+        _endBlock,
+        _rewardLockBps,
+        _startBlock
+      );
+    }
+
+    (pool.rewardPerBlock, pool.accRewardPerShare) = (_rewardPerBlock, 0);
```

Comment:
we assign pool.accRewardPerShare = 0, whether user can claim reward from old campaigns?

Reply:
yeah they won't be able to claim the old reward, since data has been overided.
It seems we won't be able to renew a pool unless we can make sure all stakers have claimed rewards